sql/delete.go

```diff
@@ -28,12 +29,20 @@ import (
 )
 
 // Delete deletes rows from a table.
+// Privileges: WRITE and READ on table. We currently always use a select statement.
+//   Notes: postgres requires DELETE. Also requires SELECT for "USING" and "WHERE" with tables.
+//          mysql requires DELETE. Also requires SELECT is a table is used in the "WHERE" clause.
 func (p *planner) Delete(n *parser.Delete) (planNode, error) {
 	tableDesc, err := p.getAliasedTableDesc(n.Table)
 	if err != nil {
 		return nil, err
 	}
 
+	if !tableDesc.HasPrivilege(p.user, parser.PrivilegeWrite) {
+		return nil, fmt.Errorf("user %s does not have %s privilege on table %s",
+			p.user, parser.PrivilegeWrite, tableDesc.Name)
```

Comment:
Actually, looks like it could be a method that returns error.  
```
if err := tableDesc.CheckPrivilege(user, privilege)
```
Added bonus: awesome function name.